substrate/client/tracing/src/logging/fast_local_time.rs

```diff
@@ -118,37 +165,53 @@ impl FormatTime for FastLocalTime {
 			}
 
 			let mut slice = cache.buffer.as_str();
-			if !self.with_fractional {
-				slice = &slice[..TIMESTAMP_PARTIAL_LENGTH];
+
+			if self.utc {
+				w.write_str(slice)?;
+			} else {
```

Comment:
The `with_fractional` should not be silently ignored when using UTC timestamps.

Either we make it obey `with_fractional`, or the timestamp format should be an enum (so then in the UTC case there just wouldn't be `with_fractional`)